x/stakeibc/keeper/lsm.go

```diff
@@ -108,3 +108,22 @@ func (k Keeper) ShouldQueryValidatorExchangeRate(ctx sdk.Context, validator type
 	// Submit query if the query interval checkpoint has been breached
 	return oldProgress.Quo(queryInterval).LT(newProgress.Quo(queryInterval))
 }
+
+// Helper function to Refund an LSM Token to the user if both:
+//  a) the LSMLiquidStake transaction is interrupted with a validator exchange rate query, and
+//  b) the handling of the query callback fails
+func (k Keeper) RefundLSMToken(ctx sdk.Context, lsmLiquidStake types.LSMLiquidStake) error {
+	liquidStakerAddress := lsmLiquidStake.Staker
+	hostZoneAddress, err := sdk.AccAddressFromBech32(lsmLiquidStake.HostZone.Address)
+	if err != nil {
+		return errorsmod.Wrapf(err, "host zone address is invalid")
+	}
+	refundToken := lsmLiquidStake.LSMIBCToken
+
+	if err := k.bankKeeper.SendCoins(ctx, hostZoneAddress, liquidStakerAddress, sdk.NewCoins(refundToken)); err != nil {
```

Comment:
Sanity checking my understanding: given the record on the tokens the user deposited is in state PENDING, they will not be ibc-transfered to the host at the top of the epoch, so we can be sure that they will be sitting in the x/stakeibc module account, right?

Reply:
close! The transfer actually happens asap (instead of epochly), but this refund is only called in the case where we have the async-query-interrupted LSM Liquid Stake - and in that case the transfer is not initiated until after the query is returned

Reply:
Transfers can fail as well which is why we have that state in the LSMDepositStatus.  You had said before that it is unlikely to transiently fail so once it fails tha particular token would likely fail again forever... would we also want to send the token back to the original holder in the case that we get a transfer fail or will we hold on to it forever?

Reply:
Sorry just seeing this. I don't think we'll want to refund that token since we've already sent the user stTokens at that point